engine/runtime/src/main/java/org/enso/interpreter/runtime/number/EnsoBigInteger.java

```diff
@@ -137,6 +138,24 @@ Type getType(@CachedLibrary("this") TypesLibrary thisLib) {
     return EnsoContext.get(thisLib).getBuiltins().number().getBigInteger();
   }
 
+  @ExportMessage
+  TriState isIdenticalOrUndefined(Object other) {
+    if (other instanceof EnsoBigInteger otherBigInt) {
+      if (value.hashCode() == otherBigInt.hashCode()) {
```

Comment:
Wrong just like [the above](https://github.com/enso-org/enso/pull/3878/files#r1064738184).

Reply:
I got rid of all the `isIdenticalOrUndefined` messages in all our classes. They were just a legacy from my very first prototype implementation. All the functionality is now contained in `HashCodeAnyNode`.